rewrite-java-core/src/test/java/HttpTraitTest.java

```diff
@@ -0,0 +1,138 @@
+import org.intellij.lang.annotations.Language;
+import org.junit.jupiter.api.Test;
+import org.openrewrite.java.ChangeMethodName;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+/**
+ * HttpTraitTest tests interface migration from com.azure.core.client.traits.HttpTrait
+ * to io.clientcore.core.models.traits.HttpTrait.
+ * Tests simple method renaming with declarative recipe.
+ * CURRENTLY UNFINISHED. All tests for HttpTrait migration will be added here.
+ * @author Annabelle Mittendorf Smith
+ */
+
+public class HttpTraitTest implements RewriteTest {
+    /**
+     * This method defines recipes used for testing.
+     * @param spec stores settings for testing environment.
+     */
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipeFromResource("/META-INF/rewrite/rewrite.yml",
+                "com.azure.rewrite.java.core.MigrateAzureCoreSamplesToAzureCoreV2");
+
+    }
+
+    /**
+     * Test that non-interface methods with the same name are not altered.
+     */
+    @Test
+    void testDoesNotChangeNonInheritedMethods() {
+        @Language("java")String noChange = "public class TestClass {\n" +
+                "    public void retryOptions() {\n" +
+                "    }\n" +
+                "    public void pipeline() {\n" +
+                "    }\n" +
+                "    public void addPolicy() {\n" +
+                "    }\n" +
+                "}\n";
+        rewriteRun(java(noChange));
+    }
+
+    /**
+     * Test simple declarative rename of:
+     * retryOptions() to httpRetryOptions()
+     * pipeline() to httpPipeline()
+     * addPolicy() to addHttpPipelinePolicy
+     */
+    @Test
+    void declarativeRenameMethodsSuccessful() {
+        @Language("java") String before = "import com.azure.core.client.traits.HttpTrait;\n" +
+                "import com.azure.core.http.HttpClient;\n" +
+                "import com.azure.core.http.HttpPipeline;\n" +
+                "import com.azure.core.http.policy.HttpLogOptions;\n" +
+                "import com.azure.core.http.policy.HttpPipelinePolicy;\n" +
+                "import com.azure.core.http.policy.RetryOptions;\n" +
+                "import com.azure.core.util.ClientOptions;\n" +
+                "\n" +
+                "public class BlankHttpTrait implements HttpTrait<BlankHttpTrait> {\n" +
```

Comment:
Users generally don't implement traits. We should test how the builder code is transformed by the recipes. For e.g. `clientBuilder.addPolicy(new HttpPipelinePolicy(){...})` is transformed to `clientBuilder.addHttpPipelinePolicy(new HttpPipelinePolicy(){...})`

Reply:
The HttpTrait recipes are currently unfinished and weren't supposed to be included in this PR (were included by mistake). Removed in commit [492daeb](https://github.com/Azure/azsdk-java-rewrite-recipes/pull/14/commits/492daeb48360470829cc54218911b9b3e1758ace)